botcraft/src/AI/Tasks/InventoryTasks.cpp

```diff
@@ -325,19 +325,58 @@ namespace Botcraft
             Position(1, 0, 0), Position(-1, 0, 0)
             });
 
-        // Check if block is air
         bool midair_placing = true;
+        // If no face specified
+        if (!face.has_value())
+        {
+            if (allow_midair_placing) // Then we don't care and default to Up
+            {
+                face = PlayerDiggingFace::Up;
+            }
+            else
+            {
+                std::vector<PlayerDiggingFace> face_candidates; // Faces next to a block
+                for (int face = 0; face < 6; face++)
+                {
+                    const Block *neighbour_block = world->GetBlock(pos + neighbour_offsets[face]);
+                    if (!neighbour_block->GetBlockstate()->IsAir())
+                    {
+                        face_candidates.push_back(static_cast<PlayerDiggingFace>(face));
+                    }
```

Comment:
You should have a scope world mutex lock to prevent block from changing between when you get the pointer and when you use it;
std::lock_guard<std::mutex> world_guard(world->GetMutex());